src/main/java/org/springframework/integration/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -326,12 +325,20 @@ public void start() {
 						Arrays.asList(partitions), offsetManager, concurrency, queueSize, dispatcherTaskExecutor,
 						autoCommitOnError);
 				this.messageDispatcher.start();
-				partitionsByBrokerMap.clear();
-				partitionsByBrokerMap.putAll(partitionsAsList.groupBy(getLeader));
+				fetchTasksByBroker.clear();
+				ImmutableListMultimap<BrokerAddress, Partition> partitionsByLeader = partitionsAsList
+						.groupBy(getLeader);
 				if (fetchTaskExecutor == null) {
-					fetchTaskExecutor = Executors.newFixedThreadPool(partitionsByBrokerMap.keysView().size());
+					fetchTaskExecutor = new SimpleAsyncTaskExecutor("kafka-fetch-");
+				}
+				if (adminTaskExecutor == null) {
+					adminTaskExecutor = Executors.newSingleThreadExecutor();
+				}
+				for (Pair<BrokerAddress, RichIterable<Partition>> entry : partitionsByLeader.keyMultiValuePairsView()) {
+					FetchTask fetchTask = new FetchTask(entry.getOne(), entry.getTwo());
+					fetchTaskExecutor.execute(fetchTask);
+					fetchTasksByBroker.put(entry.getOne(), fetchTask);
 				}
-				partitionsByBrokerMap.forEachKey(launchFetchTask);
```

Comment:
Another out of topic concern:
the `this.offsetManager` population here in the `start()`.
The `stop()` doesn't check it for `null` - just because we may not start it at all, but can call `stop()`. 
Something like regression testing :smile: